hydroflow/tests/lattice_join.rs

```diff
@@ -0,0 +1,134 @@
+use hydroflow::util::collect_ready;
+use hydroflow_macro::hydroflow_syntax;
+use multiplatform_test::multiplatform_test;
+
+#[multiplatform_test]
+pub fn test_lattice_join_reducing_behavior() {
+    // lattice_fold has this particular behavior where it uses LatticeFrom to convert the first element into the another lattice type and then it folds into that accumulator.
+    // This is to support things like SetUnionSingletonSet -> SetUnionHashSet. It also helps with things like Min, where there is not obvious default value so you want reduce-like behavior.
+    // There's kind of 3 ways you can do this reducing thing, you can do Fold(default_fn, fold_fn), Reduce(reduce_fn), and the third way which is halfway between the two previous approaches, FoldFrom(From_fn, Fold_fn).
+    // The characteristics of the three are, Fold requires a default function, Reduce requires Fn(&mut X, X) -> X, and FoldFrom requires a converter function From_fn: Fn(X) -> A, Fold_fn: Fn(&mut A, X)
+    let mut df = hydroflow_syntax! {
+        use lattices::Min;
+        use lattices::Max;
+
+        source_iter([(7, Min::new(5)), (7, Min::new(6))]) -> [0]my_join;
+        source_iter([(7, Max::new(5)), (7, Max::new(6))]) -> [1]my_join;
+
+        my_join = lattice_join::<Min<usize>, Max<usize>>()
+            -> assert_eq([(7, (Min::new(5), Max::new(6)))]); // If Min::default is used, then this would likely produce (7, (0, 6)), which is incorrect.
+    };
+
+    df.run_available();
+}
+
+#[multiplatform_test]
+pub fn test_lattice_join_set_union() {
+    let mut df = hydroflow_syntax! {
+        use lattices::set_union::SetUnionSingletonSet;
+        use lattices::set_union::SetUnionHashSet;
+
+        source_iter([(7, SetUnionHashSet::new_from([5])), (7, SetUnionHashSet::new_from([6]))]) -> [0]my_join;
+        source_iter([(7, SetUnionSingletonSet::new_from(5)), (7, SetUnionSingletonSet::new_from(6))]) -> [1]my_join;
+
+        my_join = lattice_join::<SetUnionHashSet<usize>, SetUnionHashSet<usize>>()
+            -> assert_eq([(7, (SetUnionHashSet::new_from([5, 6]), SetUnionHashSet::new_from([5, 6])))]); // If Min::default is used, then this would likely produce (7, (0, 6)), which is incorrect.
+    };
+
+    df.run_available();
+}
+
+#[multiplatform_test]
+pub fn test_lattice_join_map_union() {
+    // lattice_fold has this particular behavior where it uses LatticeFrom to convert the first element into the another lattice type and then it folds into that accumulator.
+    // This is to support things like SetUnionSingletonSet -> SetUnionHashSet. It also helps with things like Min, where there is not obvious default value so you want reduce-like behavior.
+    let mut df = hydroflow_syntax! {
```

Comment:
replicated meaty comments in test files ... seems a little odd. This is important material, should go somewhere.

Reply:
copy and paste artifacts from above. I agree it is odd, these were here before the operator docs were written, so this thing did beg for an explanation and now it has a place for it in the operator docs so I'll delete it from here.